### DISCUSSION TITLE
Argocd vault plugin not working

### DISCUSSION THREAD

**Discussion by ronnh** [argoproj/argo-cd#19945] [General]
Hi, I'm trying to setup the Argo CD Vault Plugin with our existing argocd instance, but it is not working as expected. 

I followed the instruction on this page [here](https://argocd-vault-plugin.readthedocs.io/en/stable/installation/#initcontainer-and-configuration-via-sidecar) to setup the sidecar method.

But the secret is not being replaced with the secret values. 

I tried adding the plugin section in the applicationset, but the following error came up in the avp pod.

```
time="2024-09-16T06:54:42Z" level=error msg="finished streaming call with code Unknown" error="match repository error: error running find command: `sh -c \"find . -name '*.yaml' | xargs -I {} grep \\\"<path\\\\|avp\\\\.kubernetes\\\\.io\\\" {} | grep .\"` failed exit status 1" grpc.code=Unknown grpc.method=MatchRepository grpc.service=plugin.ConfigManagementPluginService grpc.start_time="2024-09-16T06:54:42Z" grpc.time_ms=373.771 span.kind=server system=grpc
```

For the record, I copied a test secrets yaml file and exec'ed into the avp pod and manually ran the `argocd-vault-plugin generate ./` command. This worked. The secret was replaced in the output. So i'm thinking the issue lies with the discovery command. 

Any help is much appreciated, thanks.

**Comment by nastacio** [comment]
Does `find . -name '*.yaml' | xargs -I {} grep \\\"<path\\\\|avp\\\\.kubernetes\\\\.io\\\" {} | grep .` work inside that avp pod?
